octobot_trading/api/modes.py

```diff
@@ -29,20 +30,73 @@ def init_trading_mode_config(config, trading_tentacles_path) -> None:
+async def create_trading_modes(config: dict,
+                               exchange_manager: ExchangeManager,
+                               cryptocurrencies: list = None,
+                               symbols: list = None,
+                               time_frames: list = None) -> list:
+    return await _create_trading_modes(trading_mode_class=get_activated_trading_mode(config),
+                                       config=config,
+                                       exchange_manager=exchange_manager,
+                                       cryptocurrencies=cryptocurrencies,
+                                       symbols=symbols,
+                                       time_frames=time_frames)
+
+
+async def _create_trading_modes(trading_mode_class: AbstractTradingMode.__class__,
+                                config: dict,
+                                exchange_manager: ExchangeManager,
+                                cryptocurrencies: list = None,
+                                symbols: list = None,
+                                time_frames: list = None) -> list:
+    return [
+        await create_trading_mode(trading_mode_class=trading_mode_class,
+                                  config=config,
+                                  exchange_manager=exchange_manager,
+                                  cryptocurrency=cryptocurrency,
+                                  symbol=symbol,
+                                  time_frame=time_frame)
+        for cryptocurrency in __get_cryptocurrencies_to_create(trading_mode_class, cryptocurrencies)
+        for symbol in __get_symbols_to_create(trading_mode_class, symbols)
+        for time_frame in __get_time_frames_to_create(trading_mode_class, time_frames)
+    ]
+
+
+async def create_trading_mode(trading_mode_class: AbstractTradingMode.__class__,
+                              config: dict,
+                              exchange_manager: ExchangeManager,
+                              cryptocurrency: str = None,
+                              symbol: str = None,
+                              time_frame: object = None) -> AbstractTradingMode:
     try:
-        trading_mode = util_get_activated_trading_mode(config)(config, exchange_manager)
+        trading_mode: AbstractTradingMode = trading_mode_class(config, exchange_manager)
+        trading_mode.cryptocurrency = cryptocurrency
+        trading_mode.symbol = symbol
+        trading_mode.time_frame = time_frame
         await trading_mode.initialize()
-        get_logger(f"{LOGGER_TAG}[{exchange_manager.exchange.name}]")\
+        get_logger(f"{LOGGER_TAG}[{exchange_manager.exchange_name}]") \
             .debug(f"Using {trading_mode.get_name()} trading mode")
+        return trading_mode
     except RuntimeError as e:
         get_logger(LOGGER_TAG).error(e.args[0])
         raise e
+
+
+def __get_cryptocurrencies_to_create(trading_mode_class: AbstractTradingMode.__class__, cryptocurrencies: list) -> list:
```

Comment:
Why __ notation ?

Reply:
Oups, that change should be in https://github.com/Drakkar-Software/OctoBot-Trading/pull/57.
And yes I made a naming typo error :+1:

Reply:
Fixed now thanks :+1: